### ISSUE TITLE
[Performance]: Vectorize Sliced Score Matching and Remove the For Loop for Porjections

### ISSUE THREAD

**Issue by soran-ghaderi** [soran-ghaderi/torchebm#76]
### Prerequisites

- [x] I have searched existing issues to ensure this performance issue hasn't been reported before
- [x] I have profiled my code to identify the bottleneck
- [x] I have tested with the latest version of TorchEBM

### Performance Issue Description

- the SSM uses a for loop to compute the loss for n_projections. 
- it should be vectorized and batch processed

### Component

Loss Function Computation

### Benchmark Code

```python
device = ssm_loss.device
x = torch.randn(10, 2, device=device)

loss = ssm_loss(x)
```

### Device Type

Mixed CPU/GPU

### System Specifications

Ubuntu 22.04

### Profiling Results

```shell

```

### Data Scale

None

### Investigation Help

- [ ] I can provide additional profiling data
- [ ] I can test proposed optimizations
- [ ] I can share my dataset (if applicable)
- [ ] I have access to different hardware for testing

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct
